sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -61,6 +61,34 @@ private case class OracleDialect() extends JdbcDialect with SQLConfHelper with N
       } else {
         super.visitAggregateFunction(funcName, isDistinct, inputs)
       }
+
+    private def compareBlob(lhs: Expression, operator: String, rhs: Expression): String = {
+      val l = inputToSQL(lhs)
+      val r = inputToSQL(rhs)
+      val op = if (operator == "<=>") "=" else operator
+      val compare = s"DBMS_LOB.COMPARE($l, $r) $op 0"
+      if (operator == "<=>") {
+        s"(($l IS NOT NULL AND $r IS NOT NULL AND $compare) OR ($l IS NULL AND $r IS NULL))"
+      } else {
+        compare
+      }
+    }
+
+    override def build(expr: Expression): String = expr match {
```

Comment:
You just need override `visitBinaryComparison`

Reply:
Unfortunately, the signature of visitBinaryComparison is accepting everything in string which loss the type information to understand if one of the arguments is binary type.

Reply:
We can only rewrite comparison for either one of the argument is BLOB. For other cases, we have to use existing implementation.

Reply:
All tests passed, but downloading report failed. We can rerun the whole test again to clear all the checks, but it takes quite some time to finish. 
https://github.com/sunxiaoguang/spark/actions/runs/12744731853

Reply:
Got it.

Reply:
LGTM. The code override build looks good to me. But we may need to add the protected method `protected String visitBinaryComparison(String name, Expression l, Expression r)`  in `V2ExpressionSQLBuilder`.
What's your opinion?  @cloud-fan

Reply:
Great, if we agree changing the signature of visitBinaryComparison in this PR, I can make the change all together to make the code easier to maintain.

Reply:
Please wait a moment. We should change the signature in another PR if we agree on.

Reply:
Sure.